src/routes/routes_v2/data.rs

```diff
@@ -1,31 +1,94 @@
 use axum::Json;
+use error_stack::ResultExt;
+use masking::PeekInterface;
 pub mod types;
 
 use crate::{
+    crypto::keymanager::{self, KeyProvider},
     custom_extractors::TenantStateResolver,
-    error::{self, ContainerError},
+    error::{self, ContainerError, ResultContainerExt},
+    routes::data::crypto_operation,
+    storage::storage_v2::VaultInterface,
+    utils,
 };
 
 pub async fn delete_data(
-    TenantStateResolver(_tenant_app_state): TenantStateResolver,
-    Json(_request): Json<types::DeleteDataRequest>,
+    TenantStateResolver(tenant_app_state): TenantStateResolver,
+    Json(request): Json<types::DeleteDataRequest>,
 ) -> Result<Json<types::DeleteDataResponse>, ContainerError<error::ApiError>> {
-    // need handle this once the key manger service is ready
-    todo!()
+    let _entity = keymanager::get_dek_manager()
+        .find_by_entity_id(&tenant_app_state, request.entity_id.clone())
+        .await?;
+
+    let _delete_status = tenant_app_state
```

Comment:
what's the `_delete_status`? can we put a assertion for this to make absolutely sure that the deletion happened?